lib/audience/api/client/gdpr.rb

```diff
@@ -0,0 +1,18 @@
+module Audience
+  module Api
+    class Client
+      module Gdpr
+        ENDPOINT = "gdpr".freeze
+        # Performs a gdpr anonymisation for the given email address
+        #
+        # Example:
+        #   Audience::Api.gdpr_anonymize("email@example.com")
+        #
+        def gdpr_anonymize(email)
+          response = delete "#{ENDPOINT}/anonymize/?email=#{email}"
```

Comment:
Hmm, didn't this accept multiple emails?

Reply:
Somehow using `DELETE` here seems weird. Are we deleting an anonymization? I feel like the path reads as we're "creating an anonymization job" so would expect it to be `POST`.